proxy/middleware.go

```diff
@@ -45,10 +46,10 @@ func (p *Proxy) redirectToSignin(w http.ResponseWriter, r *http.Request) error {
 	return nil
 }
 
-func (p *Proxy) isAuthorized(w http.ResponseWriter, r *http.Request) (bool, error) {
+func (p *Proxy) isAuthorized(w http.ResponseWriter, r *http.Request) (bool, bool, error) {
```

Comment:
The `(bool. bool, error)` is the sign that we should refactor this function more. How about introduce new return type, like:

```
type authResponse struct {
    authorized bool
    statusCode int
}
```

Then you can check status code in the caller to decide to clear the session or not.

Reply:
I'm new to Go, my day-to-day language is C#... so not used to multiple return values.  The method you suggest (i.e. returning a struct/class) would be what I would do in C# because only one return type is available.  Stylistically (both in general and for this particular project), is it preferable to do this approach, or have multiple returns?

Reply:
> Please add a test, so we can make sure the change won't introduce any regression.

From what I'm reading (correct me if I'm wrong), I don't see any tests for when the check result is not ok, so seems like a bigger issue than just this particular change.  Should these tests be added as part of this change, or should there be a separate pull to add more coverage overall?

Reply:
> I'm new to Go, my day-to-day language is C#... so not used to multiple return values. The method you suggest (i.e. returning a struct/class) would be what I would do in C# because only one return type is available. Stylistically (both in general and for this particular project), is it preferable to do this approach, or have multiple returns?

The thing is that if others look at the method signature, the `(bool, bool, error)` make no sense what it returns. If it's only `isAuthorized(w, req) (bool, error)`, then you can see that the `bool` indicated whether we are authorized or not. While with `isAuthorized(w, req) (bool, bool, error)`, you don't know for sure which one indicated authorization. Even if you know the first one is, then what the second return value mean.

What I propose is change it to `authorized(w, req) (*authRespone, error)`, then you can do what ever you want with `*authResponse`, and make the code much readable.

Reply:
Modified per suggestion.  I did not have time to figure out how to write a unit test for it, perhaps over the next week I can try to figure that out, but not sure work will allow.